commit bf7f1e9dd406c1bb36bc02a79d185560c3f8f890
Author: Alexey Kardashevskiy <aik@ozlabs.ru>
Date:   Fri Mar 17 00:48:27 2017 +0000

    vfio/spapr: Reference mm in tce_container
    
    [ Upstream commit bc82d122ae4a0e9f971f13403995898fcfa0c09e ]
    
    In some situations the userspace memory context may live longer than
    the userspace process itself so if we need to do proper memory context
    cleanup, we better have tce_container take a reference to mm_struct and
    use it later when the process is gone (@current or @current->mm is NULL).
    
    This references mm and stores the pointer in the container; this is done
    in a new helper - tce_iommu_mm_set() - when one of the following happens:
    - a container is enabled (IOMMU v1);
    - a first attempt to pre-register memory is made (IOMMU v2);
    - a DMA window is created (IOMMU v2).
    The @mm stays referenced till the container is destroyed.
    
    This replaces current->mm with container->mm everywhere except debug
    prints.
    
    This adds a check that current->mm is the same as the one stored in
    the container to prevent userspace from making changes to a memory
    context of other processes.
    
    DMA map/unmap ioctls() do not check for @mm as they already check
    for @enabled which is set after tce_iommu_mm_set() is called.
    
    This does not reference a task as multiple threads within the same mm
    are allowed to ioctl() to vfio and supposedly they will have same limits
    and capabilities and if they do not, we'll just fail with no harm made.
    
    Signed-off-by: Alexey Kardashevskiy <aik@ozlabs.ru>
    Acked-by: Alex Williamson <alex.williamson@redhat.com>
    Reviewed-by: David Gibson <david@gibson.dropbear.id.au>
    Signed-off-by: Michael Ellerman <mpe@ellerman.id.au>
    Signed-off-by: Sasha Levin <alexander.levin@verizon.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Index: linuxtest/dummy/rpi_1436_bf7f1e9dd406c1bb36bc02a79d185560c3f8f890.txt
===================================================================
--- /dev/null
+++ linuxtest/dummy/rpi_1436_bf7f1e9dd406c1bb36bc02a79d185560c3f8f890.txt
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
